src/components/Planner/Planner.tsx

```diff
@@ -46,40 +134,316 @@ const Planner = () => {
+      if (!destMealSlot || !destDayId || !sourceDayId) {
+        toast({
+          title: "You have missed the slot!",
+          description: "Drop into the slot, please",
+          variant: "subtle",
+          status: "warning",
+          duration: 3000,
+          isClosable: true,
+          position: "top"
+        });
+        return;
+      }
+
+      if (!mealExistsInSlot) {
+        movedItem.mealSlot = destMealSlot;
+        if (sourceDayId === destDayId) {
+          sourceDay.splice(destination.index, 0, movedItem);
+
+          setDays(prevDays => ({
+            ...prevDays,
+            [sourceDayId]: {
+              ...prevDays[sourceDayId],
+              recipes: sourceDay
+            }
+          }));
+        } else {
+          destinationDay.splice(destination.index, 0, movedItem);
+
+          setDays(prevDays => ({
+            ...prevDays,
+            [sourceDayId]: {
+              ...prevDays[sourceDayId],
+              recipes: sourceDay
+            },
+            [destDayId]: {
+              ...prevDays[destDayId],
+              recipes: destinationDay
+            }
+          }));
+        }
+
+        const data = {
+          _id: movedItem.mealId,
+          recipeId: movedItem.id,
+          dayOfWeek: destDayId,
+          mealSlot: movedItem.mealSlot
+        };
+
+        updateMealPlan(data)
+          .then(response => {
+            const updatedMealPlan = response.data; 
+
+            setDays(prevDays => {
+              const updatedSourceDay = [...prevDays[sourceDayId]?.recipes];
+              const updatedDestinationDay = [...prevDays[destDayId]?.recipes];
+
+              const sourceDayIndex = updatedSourceDay.findIndex(
+                item => item.mealSlot === sourceMealSlot
+              );
+              if (sourceDayIndex !== -1) {
+                updatedSourceDay.splice(sourceDayIndex, 1);
+              }
+
+              const destinationDayIndex = updatedDestinationDay.findIndex(
+                item => item.mealSlot === destMealSlot
+              );
+              if (destinationDayIndex === -1) {
+                updatedDestinationDay.push(updatedMealPlan);
+              }
+
+              return {
+                ...prevDays,
+                [sourceDayId]: {
+                  ...prevDays[sourceDayId],
+                  recipes: updatedSourceDay
+                },
+                [destDayId]: {
+                  ...prevDays[destDayId],
+                  recipes: updatedDestinationDay
+                }
+              };
+            });
+          })
+          .catch(error => {
+            showErrorToast(error);
+          });
+      } else if (mealExistsInSlot) {
+        toast({
+          title: "You cannot replace meals!",
+          description: "Delete the meal and add a new one, please",
+          variant: "subtle",
+          status: "warning",
+          duration: 3000,
+          isClosable: true,
+          position: "top"
+        });
+      } 
```

Comment:
NON-BLOCKING: You use this notification multiple times, so you could put it in an object that you then pass to `toast` each time you need it.